### DISCUSSION TITLE
Support for long-running command output?

### DISCUSSION THREAD

**Discussion by allComputableThings** [carlmontanari/scrapli#212] [Q&A]
Is it possible to incrementally receive command output. 
Suppose I wanted to run: 'tail -f /var/log/log.log'
... the commend will never complete. Is there something like `await conn.read()` to wait on the socket for more data?

Thanks in advance.

**Comment by carlmontanari** [comment]
Hey @stuz5000 

Yep, TL;DR is disable transport timeout by setting that value to 0. You can then call as many `channel.read` as you want in a while loop or whatever you need to do. Something similar to:

```python
        output = b""
        while True:
            new_output = await self.conn.channel.read()
            output += new_output
            if read_until in output:
                return

            await asyncio.sleep(1)
```

Or obviously just keep reading or do whatever you want!

You may not need to disable the transport timeout -- it governs the actual individual read operations... offhand I think its like 30s default or something, so if you never have any times where the channel goes more than 30s without output it wont matter, but probably a good idea to disable it for things like what you are talking about.

**Comment by allComputableThings** [comment]
Isn't there a solution that can await on IO instead of using await sleep in
a loop and using consuming CPU?
I think one of the intended benefits of switching to async/await in the
first place is to remove the need for this (and also remove the complexity
of epoll/select).

Thanks,
- Stuart

**Reply by carlmontanari** [reply]
channel.read does exactly what you're saying. you can chose to sleep or not, just an example.

**Comment by allComputableThings** [comment]
Oh -- I misread and didn't see that it was *await* self.conn.channel.read.
Thank you.

**Reply by carlmontanari** [reply]
No worries. Holler if that doesnt do what ya want and we can try to see if we can figure it out!
